ncurses: fix splicing

kjeremy opened the pull request:
Fixes warnings with #263082 applied

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

alyssais committed:
nixos/qemu-vm: fix infinite recursion

The virtualisation.directBoot.initrd option was added for netboot
images, but the assertion to check directBoot enabled if it was used
caused an infinite recursion if it was.  Minimal reproduction:

	import nixos/tests/make-test-python.nix ({ pkgs, ... }: {
	  name = "";

	  nodes = {
	    machine = { config, ...}: {
	      imports = [ nixos/modules/installer/netboot/netboot-minimal.nix ];

	      virtualisation.directBoot = {
	        enable = true;
	        initrd = "${config.system.build.netbootRamdisk}/${config.system.boot.loader.initrdFile}";
	      };
	    };
	  };
	  testScript = "";
	}) {}

The fix is to swap the two conditions, so that cfg.directBoot.enable
is checked first, and the initrd comparision will be short circuited.

This wasn't noticed during review because in earlier versions of the
virtualisation.directBoot patch, the assertion was accidentally in the
conditional above, so wasn't evaluated unless port forwarding was in
use.

nicoonoclaste committed:
nixos/vagrant-guest: Set `security.sudo-rs.wheelNeedsPassword` too

vbgl committed:
ocamlPackages.sodium: disable for OCaml ≥ 5.0

vbgl committed:
ocamlPackages.camlimages: fix build with OCaml ≥ 5.0

vbgl committed:
ocamlPackages.cry: fix build with OCaml ≥ 5.0

vbgl committed:
ocamlPackages.piqi: disable for OCaml ≥ 5.0

alyssais committed:
nixos/stage-1: create initramfs /lib at build time

Modules built in to the kernel can attempt to load firmware before
init is started.  To guarantee the firmware is accessible to them
where they expect, /lib has to exist in the initramfs — it can't be
created later by init, because by that point the module may already
have tried and given up.

lblasc committed:
lua-rtoml: init 0.2

vbgl committed:
ocamlPackages.dum: fix build with OCaml ≥ 5.0

Munksgaard committed:
smlfmt: 1.0.0 -> 1.1.0

kjeremy committed:
ncurses: fix splicing

Fixes warnings with #263082 applied

Update pkgs/development/libraries/ncurses/default.nix

Co-authored-by: Artturi <Artturin@artturin.com>

ncurses: Explicitly use buildPackages for --with-build-cc

stdenv.cc worked for me because my nixos was configured to use boot.binfmt.emulatedSystems.

999eagle committed:
nixos/matrix-sliding-sync: add dependency on matrix-synapse if running locally and restart

alyssais commented on pkgs/development/libraries/ncurses/default.nix:
```diff
@@ -48,20 +48,17 @@ stdenv.mkDerivation (finalAttrs: {
         "/run/current-system/sw/share/terminfo" # NixOS
       ]}"
-  ];
+  ] ++ lib.optional (stdenv.buildPlatform != stdenv.hostPlatform) "--with-build-cc=${stdenv.cc.targetPrefix}cc";
```
Wouldn't this be the host C compiler?  The build compiler would be just cc.  (Or is this a terminology mismatch between Nixpkgs and ncurses?)

kjeremy replied:
It's definitely confusing. See https://github.com/mirror/ncurses/blob/master/INSTALL#L2421

alyssais replied:
Could maybe use a comment then.

Artturin replied:
> actually had it set to cc and it failed during configure

because there's no cc unless the for build cc is in depsBuildBuild

Artturin replied:
and indeed `pkgsCross.aarch64-multiplatform.ncurses` fails

```
ncurses-aarch64-unknown-linux-gnu> checking if the build-compiler "aarch64-unknown-linux-gnu-cc" works... no
```

kjeremy replied:
I pulled that out of depsBuildBuild to get rid of the mismatch warning so I'm not sure what the fix would be.

kjeremy replied:
Oh! My nixos install had binfmt enabled and that's why it succeeded for me.

Artturin commented on pkgs/development/libraries/ncurses/default.nix:
```diff
@@ -48,20 +48,17 @@ stdenv.mkDerivation (finalAttrs: {
         "/run/current-system/sw/share/terminfo" # NixOS
       ]}"
-  ];
+  ] ++ lib.optional (stdenv.buildPlatform != stdenv.hostPlatform) "--with-build-cc=${stdenv.cc.targetPrefix}cc";
```
This is wrong, build cc is for build so this targetPrefix is wrong.

Artturin commented on pkgs/development/libraries/ncurses/default.nix:
```diff
@@ -48,20 +48,17 @@ stdenv.mkDerivation (finalAttrs: {
     pkg-config
   ] ++ lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [
-    buildPackages.ncurses
+    ncurses
```
```suggestion
   # for `tic`, build already depends on for build `cc` so it's weird the build doesn't just build `tic`.
    ncurses
```

kjeremy commented:
Changed to target `staging`

AndersonTorres committed:
xosview: migrate to by-name

AndersonTorres committed:
xosview2: migrate to by-name

AndersonTorres committed:
xosview: set meta.mainProgram

AndersonTorres committed:
xosview: split man output

AndersonTorres committed:
xosview2: set meta.mainProgram

AndersonTorres committed:
xosview2: 2.3.2 -> 2.3.3

AndersonTorres committed:
uxn: unstable-2023-09-29 -> unstable-2023-10-23

chewblacka committed:
clamtk: init at 6.16

r-ryantm committed:
gdal: 3.7.2 -> 3.7.3

pluiedev committed:
nixos/greetd: autostart GNOME Keyring when enabled

The PAM service `greetd` creates now autostarts GNOME Keyring when it is
enabled via `services.gnome.gnome-keyring.enable`.

Should cause a lot fewer headaches for people like me who didn't know
why GNOME Keyring was complaining that it was never supplied with login details.

Fixes #246197

ipetkov committed:
syncoid: disable PrivateUsers in systemd unit

* leaving PrivateUsers=true appears to break when using zfs-2.2.0

numinit committed:
androidenv: updates for Android API 34

r-ryantm committed:
numcpp: 2.12.0 -> 2.12.1

natsukium committed:
python311Packages.x11-hash: rename from x11_hash

natsukium committed:
python311Packages.x11-hash: refactor

fabaff committed:
python311Packages.debianbts: refactor

fabaff committed:
python310Packages.derpconf: add changelog to meta

Kranzes committed:
Merge pull request #265601 from r-ryantm/auto-update/python310Packages.cryptolyzer

python310Packages.cryptolyzer: 0.10.0 -> 0.10.3

fabaff committed:
python310Packages.derpconf: add pythonImportsCheck

Kranzes committed:
Merge pull request #265600 from r-ryantm/auto-update/python310Packages.cryptoparser

python310Packages.cryptoparser: 0.10.3 -> 0.11.0

fabaff committed:
python310Packages.derpconf: add format

- disable on unsupported Python releases

r-ryantm committed:
python310Packages.diffsync: 1.8.0 -> 1.9.0

Commit:
vencord: 1.6.2 -> 1.6.3

marsam committed:
Merge pull request #264925 from marsam/update-docker-slim

docker-slim: 1.40.4 -> 1.40.5

marsam committed:
Merge pull request #265492 from r-ryantm/auto-update/postgresql12JitPackages.plpgsql_check

postgresql12JitPackages.plpgsql_check: 2.5.4 -> 2.6.0

marsam committed:
Merge pull request #265586 from marsam/update-gallery-dl

gallery-dl: 1.26.1 -> 1.26.2

marsam committed:
Merge pull request #265567 from r-ryantm/auto-update/nerdctl

nerdctl: 1.6.2 -> 1.7.0

marsam committed:
Merge pull request #265456 from foo-dogsquared/update-decker

decker: 1.31 -> 1.32

RaitoBezarius committed:
Merge pull request #265281 from JulienMalka/systemd-boot-maintainer

marsam committed:
Merge pull request #265305 from r-ryantm/auto-update/procs

procs: 0.14.0 -> 0.14.3

marsam committed:
Merge pull request #265219 from r-ryantm/auto-update/kubectl-gadget

kubectl-gadget: 0.21.0 -> 0.22.0

r-ryantm committed:
python310Packages.django-admin-sortable2: 2.1.9 -> 2.1.10

davidkna committed:
cargo-outdated: add missing `CoreServices` buildinput

marsam committed:
Merge pull request #265484 from adamcstephens/master

gickup: 0.10.21 -> 0.10.22

marsam committed:
Merge pull request #264436 from anthonyroussel/updatess/meritous

meritous: 1.4 -> 1.5

marsam committed:
Merge pull request #265485 from adamcstephens/mainprogram/shairport-nqptp

shairport-sync, nqptp: add meta.mainProgram and maintainer

marsam committed:
Merge pull request #265403 from YorikSar/dark-mode-notify-mainProgram

dark-mode-notify: add meta.mainProgram

flokli committed:
Merge pull request #265595 from r-ryantm/auto-update/ls-lint

ls-lint: 2.1.0 -> 2.2.2

fabaff committed:
Merge pull request #265631 from fabaff/vine-bump

python311Packages.vine: 5.0.0 -> 5.1.0

markuskowa committed:
nwchem: 7.2.1 -> 7.2.2

fabaff committed:
Merge pull request #265626 from r-ryantm/auto-update/python310Packages.debianbts

python310Packages.debianbts: 4.0.1 -> 4.0.2

fabaff committed:
Merge pull request #265629 from r-ryantm/auto-update/python310Packages.derpconf

python310Packages.derpconf: 0.8.3 -> 0.8.4

fabaff committed:
Merge pull request #265605 from r-ryantm/auto-update/python310Packages.cx-freeze

python310Packages.cx-freeze: 6.15.7 -> 6.15.10

fabaff committed:
python311Packages.django-redis: ignore DeprecationWarning

K900 committed:
Merge pull request #257551 from toastal/v0.0.20230924

lunarml: unstable-2023-09-21 → 0.0.20230924

aanderse committed:
Merge pull request #256738 from toastal/prosody-libevent-deprecated

prosody: remove deprecated libevent

fabaff committed:
python311Packages.django-redis: migrate to pyproject

prusnak committed:
Merge pull request #265635 from erikarvstedt/electrs-0.10.1

electrs: 0.9.13 -> 0.10.1

NickCao committed:
Merge pull request #265632 from r-ryantm/auto-update/python310Packages.diagrams

python310Packages.diagrams: 0.23.3 -> 0.23.4

NickCao committed:
Merge pull request #265638 from fwam/vencord

vencord: 1.6.2 -> 1.6.3

NickCao committed:
Merge pull request #265639 from r-ryantm/auto-update/python310Packages.diffsync

python310Packages.diffsync: 1.8.0 -> 1.9.0

fabaff committed:
python311Packages.aiosql: adjust inputs

NickCao committed:
Merge pull request #265644 from r-ryantm/auto-update/python310Packages.django-admin-sortable2

python310Packages.django-admin-sortable2: 2.1.9 -> 2.1.10

NickCao committed:
Merge pull request #265617 from mimame/update-tailspin-2.0.0

tailspin: 1.6.0 -> 2.0.0

NickCao committed:
Merge pull request #265602 from r-ryantm/auto-update/python310Packages.css-parser

python310Packages.css-parser: 1.0.9 -> 1.0.10

NickCao committed:
Merge pull request #265599 from natsukium/pylyzer/update

pylyzer: 0.0.48 -> 0.0.49

NickCao committed:
Merge pull request #265598 from r-ryantm/auto-update/phrase-cli

phrase-cli: 2.12.0 -> 2.15.0

NickCao committed:
Merge pull request #265591 from r-ryantm/auto-update/python310Packages.crate

python310Packages.crate: 0.33.0 -> 0.34.0

NickCao committed:
Merge pull request #265590 from r-ryantm/auto-update/python310Packages.correctionlib

python310Packages.correctionlib: 2.3.3 -> 2.4.0

NickCao committed:
Merge pull request #265587 from natsukium/proxy-tools

python311Packages.proxy-tools: rename from proxy_tools

NickCao committed:
Merge pull request #265578 from r-ryantm/auto-update/python310Packages.clickhouse-connect

python310Packages.clickhouse-connect: 0.6.11 -> 0.6.18

NickCao committed:
Merge pull request #265577 from r-ryantm/auto-update/python310Packages.click-odoo-contrib

python310Packages.click-odoo-contrib: 1.17.0 -> 1.18.0

NickCao committed:
Merge pull request #265540 from r-ryantm/auto-update/libsForQt5.qtpbfimageplugin

libsForQt5.qtpbfimageplugin: 2.5 -> 2.6

NickCao committed:
Merge pull request #265499 from wineee/wvkbd

wvkbd: 0.14.1 -> 0.14.3

fabaff committed:
python311Packages.django-q: mark as broken

drupol committed:
vscode-extensions.griimick.vhs: init at 0.0.4 (#265634)

* vscode-extensions.griimick.vhs: init at 0.0.4

* Update pkgs/applications/editors/vscode/extensions/default.nix

Co-authored-by: superherointj <5861043+superherointj@users.noreply.github.com>

---------

Co-authored-by: superherointj <5861043+superherointj@users.noreply.github.com>

onny committed:
nixos/printing: Add openFirewall option (#176539)

r-ryantm committed:
python311Packages.python-smarttub: 0.0.34 -> 0.0.35

r-ryantm committed:
python311Packages.python-on-whales: 0.65.0 -> 0.66.0

fabaff committed:
Merge pull request #265662 from fabaff/aiosql-fix

python311Packages.aiosql: adjust inputs

Atemu committed:
Revert "btrfs-progs: 6.5.3 -> 6.6"

This reverts commit a6c6e32d7fd7223eee4a6d86a452a34bbcc403ae.

fabaff committed:
Merge pull request #265655 from fabaff/baserow-fix

python311Packages.django-redis: ignore DeprecationWarning

symphorien committed:
Merge pull request #265131 from onemoresuza/aba

aba: 0.7.0 -> 0.7.1

symphorien committed:
Merge pull request #265026 from figsoda/felix

felix-fm: 2.9.0 -> 2.10.1

r-ryantm committed:
sqlite-utils: 3.35.1 -> 3.35.2

r-ryantm committed:
tbox: 1.7.4 -> 1.7.5

notalltim committed:
grpc: explicitly use the build platform for `grpc_cpp_plugin`
when cross building grpc use the build platform `grpc_cpp_plugin` to generate the internal protos

Atemu committed:
Merge pull request #265677 from Atemu/revert/trofi/btrfs-progs-update

Revert "btrfs-progs: 6.5.3 -> 6.6"

al3xtjames committed:
python311Packages.img2pdf: fix evaluation on darwin

The img2pdf derivation uses the sRGB ICC profile provided by colord
starting with #264119. colord and several of its dependencies only
support Linux, so this broke evaluation on Darwin. This change drops the
use of colord on Darwin to fix the derivation; the system-provided sRGB
profile is used instead.

r-ryantm committed:
wasmedge: 0.13.4 -> 0.13.5

dotlambda committed:
Merge pull request #263967 from r-ryantm/auto-update/python310Packages.imapclient

python310Packages.imapclient: 2.3.1 -> 3.0.0

dotlambda committed:
Merge pull request #264945 from dotlambda/bitwarden-2023.10.0

bitwarden: 2023.9.3 -> 2023.10.0

fabaff committed:
Merge pull request #264185 from fabaff/msal-bump

python311Packages.msal: 1.24.0 -> 1.24.1

fabaff committed:
Merge pull request #265669 from r-ryantm/auto-update/python311Packages.python-smarttub

python311Packages.python-smarttub: 0.0.34 -> 0.0.35

K900 committed:
Merge pull request #265487 from K900/vscode-1.84

vscode, vscode-server: 1.83.1 -> 1.84.0

K900 committed:
Merge pull request #264207 from K900/mitmproxy-10

mitmproxy: 9.0.1 -> 10.1.1

K900 committed:
Merge pull request #265411 from K900/fix-all-the-uboots

uboot: a whole bunch of build fixes

K900 committed:
nix-prefetch-git: download submodules with --progress

K900 committed:
yuzu: 1595 -> 1611, yuzu-ea: 3940 -> 3966

r-ryantm committed:
libgpiod: 2.0.2 -> 2.1

r-ryantm committed:
kodiPackages.pvr-hts: 20.6.3 -> 20.6.4

lilyinstarlight committed:
Merge pull request #265579 from muscaln/teams-for-linux

teams-for-linux: 1.3.14 -> 1.3.18

krupkat committed:
xpano: 0.16.1 -> 0.17.0

r-ryantm committed:
python310Packages.django-cacheops: 7.0.1 -> 7.0.2

K900 committed:
Merge pull request #262184 from aidalgol/missing-game-fhsenv-libs

Add libunwind to Heroic and Steam FHS envs

Stunkymonkey committed:
nixos/prometheus.exporters.knot: migrate from extraConfig to settingsFile

Stunkymonkey committed:
nixos/prometheus.exporters.pgbouncer: migrate from connectionString to connectionStringFile

r-ryantm committed:
python310Packages.django-configurations: 2.4.1 -> 2.5

toonn committed:
Merge pull request #262659 from reckenrode/crystal-fix

crystal: fix build with newer versions of clang

lunik1 committed:
iosevka: 27.3.2 -> 27.3.4

Diff: https://github.com/be5invis/iosevka/compare/v27.3.2...v27.3.4

marsam committed:
Merge pull request #265592 from marsam/update-mystmd

mystmd: 1.1.23 -> 1.1.26

marsam committed:
Merge pull request #265637 from scm2342/kubectl-klock-0.5.0

kubectl-klock: 0.4.0 -> 0.5.0

marsam committed:
Merge pull request #265266 from R-VdP/rvdp/fwupd_1_9_7

fwupd: 1.9.6 -> 1.9.7

xworld21 committed:
ghostscript: pad headers at link time to prevent install_name_tool failure on Darwin (#263833)

r-ryantm committed:
python310Packages.django-crispy-forms: 2.0 -> 2.1

dotlambda committed:
Merge pull request #265501 from r-ryantm/auto-update/python310Packages.wand

python310Packages.wand: 0.6.11 -> 0.6.13

r-ryantm committed:
python310Packages.django-ipware: 5.0.0 -> 5.0.2

dywedir committed:
Merge pull request #265643 from davidkna/cargo-outdated-coreservices

cargo-outdated: add missing `CoreServices` buildinput

999eagle committed:
nixos/matrix-synapse: add readOnly serviceUnit option

This is useful to configure services dependent on matrix-synapse regardless
of whether workers are enabled or not.

999eagle committed:
nixos/matrix/*: change dependencies on matrix-synapse.service to serviceUnit

Atemu committed:
Merge pull request #265181 from zzzsyyy/update/xanmod

linux_xanmod, linux_xanmod_latest: 2023-11-02

K900 committed:
Merge pull request #265702 from K900/fresh-fruit

yuzu: 1595 -> 1611, yuzu-ea: 3940 -> 3966

adisbladis committed:
Merge pull request #258299 from Arcayr/virtualisation-hostname

nixos/virtualisation: add hostname option to oci-containers.

adisbladis committed:
Merge pull request #265585 from marsam/add-emacsPackages.lspce

emacsPackages.lspce: init at unstable-2023-10-30

jbedo committed:
Merge pull request #261763 from nviets/nngUpdate

nng: 1.6.0-prerelease update for rPackages.nanonext

Ma27 committed:
Merge pull request #265708 from Stunkymonkey/nixos-test-prometheus-exporters

nixos/prometheus-exporters: fix knot & pgbounce tests

Ma27 committed:
Merge pull request #265071 from 999eagle/feat/matrix-sliding-sync-update

nixos/matrix-sliding-sync: add dependency on matrix-synapse if running locally

uri-canva committed:
Merge pull request #265130 from r-ryantm/auto-update/opentelemetry-collector

opentelemetry-collector: 0.87.0 -> 0.88.0

r-ryantm committed:
python310Packages.djangorestframework-dataclasses: 1.3.0 -> 1.3.1

aanderse committed:
Merge pull request #265705 from r-ryantm/auto-update/kodiPackages.pvr-hts

kodiPackages.pvr-hts: 20.6.3 -> 20.6.4

markuskowa committed:
Merge pull request #265649 from markuskowa/upd-nwchem

nwchem: 7.2.1 -> 7.2.2

NickCao committed:
Merge pull request #265682 from r-ryantm/auto-update/tbox

tbox: 1.7.4 -> 1.7.5

NickCao committed:
Merge pull request #265681 from r-ryantm/auto-update/sqlite-utils

sqlite-utils: 3.35.1 -> 3.35.2

NickCao committed:
Merge pull request #265670 from r-ryantm/auto-update/python311Packages.python-on-whales

python311Packages.python-on-whales: 0.65.0 -> 0.66.0

natsukium committed:
Merge pull request #265575 from natsukium/zxing-cpp

python311Packages.zxing-cpp: rename from zxing_cpp

jfvillablanca committed:
llm-ls: init at 0.4.0

r-ryantm committed:
python310Packages.duckduckgo-search: 3.8.5 -> 3.9.4

r-ryantm committed:
python310Packages.dvclive: 3.1.0 -> 3.2.0

Artturin committed:
Merge pull request #265570 from amjoseph-nixpkgs/pyqtwebengine-take-different-python3

dotlambda committed:
Merge pull request #264418 from dotlambda/restic-0.16.2

restic: 0.16.1 -> 0.16.2

AndersonTorres committed:
sxhkd: refactor

- migrate to by-name
- split outputs
- add meta.mainProgram

r-ryantm committed:
python310Packages.easydict: 1.10 -> 1.11

dotlambda committed:
Merge pull request #265752 from al3xtjames/img2pdf-icc-path

python311Packages.img2pdf: fix evaluation on darwin

amjoseph-nixpkgs commented on pkgs/development/libraries/ncurses/default.nix:
```diff
@@ -48,20 +48,18 @@ stdenv.mkDerivation (finalAttrs: {
         "/run/current-system/sw/share/terminfo" # NixOS
       ]}"
-  ];
+  ] ++ lib.optional (stdenv.buildPlatform != stdenv.hostPlatform) "--with-build-cc=${stdenv.cc.targetPrefix}cc";
```
```suggestion

  depsBuildBuild = [ stdenv ];
```

amjoseph-nixpkgs replied:
`stdenv` is spliced so it works here.  Unfortunately `stdenv.cc` isn't spliced :frowning:

amjoseph-nixpkgs replied:
If you make this change I will close https://github.com/NixOS/nixpkgs/pull/265759/files and merge this instead (after the 23.11 branch-off for `staging`, which happens in about a week)

Artturin replied:
https://github.com/NixOS/nixpkgs/pull/265162#discussion_r1382748821

amjoseph-nixpkgs replied:
Nevermind.

kashw2 committed:
terragrunt: 0.53.0 -> 0.53.2

r-ryantm committed:
python310Packages.ezyrb: 1.3.0.post2309 -> 1.3.0.post2311

samueldr committed:
Merge pull request #262684 from samueldr/feature/nixos-rebuild-completions

nixos-rebuild: Locally own the nixos-rebuild (bash) completions

r-ryantm committed:
python310Packages.faster-whisper: 0.8.0 -> 0.9.0

r-ryantm committed:
python310Packages.fasttext-predict: 0.9.2.1 -> 0.9.2.2

r-ryantm committed:
python310Packages.finvizfinance: 0.14.6 -> 0.14.7

marsam committed:
Merge pull request #260727 from bcdarwin/opencv-python-updates

python310Packages.opencv3: remove

fabianhjr committed:
Merge pull request #265720 from r-ryantm/auto-update/python310Packages.django-crispy-forms

python310Packages.django-crispy-forms: 2.0 -> 2.1

fabianhjr committed:
Merge pull request #265715 from lunik1/iosevka-27.3.4

iosevka: 27.3.2 -> 27.3.4

marsam committed:
Merge pull request #265758 from r-ryantm/auto-update/python310Packages.duckduckgo-search

python310Packages.duckduckgo-search: 3.8.5 -> 3.9.4

marsam committed:
Merge pull request #265749 from r-ryantm/auto-update/python310Packages.djangorestframework-dataclasses

python310Packages.djangorestframework-dataclasses: 1.3.0 -> 1.3.1

fabianhjr committed:
Merge pull request #265553 from r-ryantm/auto-update/ocserv

ocserv: 1.2.1 -> 1.2.2

fabianhjr committed:
Merge pull request #265539 from kilianar/portfolio-0.65.5

portfolio: 0.65.4 -> 0.65.5

marsam committed:
Merge pull request #265723 from r-ryantm/auto-update/python310Packages.django-ipware

python310Packages.django-ipware: 5.0.0 -> 5.0.2

marsam committed:
Merge pull request #265691 from r-ryantm/auto-update/wasmedge

wasmedge: 0.13.4 -> 0.13.5

marsam committed:
Merge pull request #265684 from notalltim/grpc-cross-build-cpp-plugin

grpc: explicitly use the build platform for `grpc_cpp_plugin`

marsam committed:
Merge pull request #265593 from r-ryantm/auto-update/carapace

carapace: 0.28.2 -> 0.28.3

marsam committed:
Merge pull request #265594 from r-ryantm/auto-update/gitoxide

gitoxide: 0.30.0 -> 0.31.1

marsam committed:
Merge pull request #265581 from r-ryantm/auto-update/python310Packages.cobs

python310Packages.cobs: 1.2.0 -> 1.2.1

r-ryantm committed:
python310Packages.flask-paginate: 2023.10.8 -> 2023.10.24

r-ryantm committed:
python310Packages.fschat: 0.2.30 -> 0.2.32

dotlambda committed:
Merge pull request #265498 from r-ryantm/auto-update/python310Packages.gdal

gdal: 3.7.2 -> 3.7.3

r-ryantm committed:
python310Packages.glcontext: 2.4.0 -> 2.5.0

honnip committed:
pretendard: 1.3.8 -> 1.3.9

r-ryantm committed:
python310Packages.gocardless-pro: 1.47.0 -> 1.48.0

K900 committed:
Merge pull request #265700 from K900/nix-prefetch-git-progress

nix-prefetch-git: download submodules with --progress

fabaff committed:
python311Packages.alexapy: 1.27.7 -> 1.27.8

Diff: https://gitlab.com/keatontaylor/alexapy/-/compare/refs/tags/v1.27.7...v1.27.8

Changelog: https://gitlab.com/keatontaylor/alexapy/-/blob/v1.27.8/CHANGELOG.md

fabaff committed:
python311Packages.boschshcpy: 0.2.73 -> 0.2.75

Diff: https://github.com/tschamm/boschshcpy/compare/0.2.73...0.2.75

fabaff committed:
python311Packages.amqp: 5.1.1 -> 5.2.0

fabaff committed:
python311Packages.dvclive: 3.1.0 -> 3.2.0

Diff: https://github.com/iterative/dvclive/compare/refs/tags/3.1.0...3.2.0

Changelog: https://github.com/iterative/dvclive/releases/tag/3.2.0

fabaff committed:
python311Packages.aiocomelit: 0.3.0 -> 0.3.1

Diff: https://github.com/chemelli74/aiocomelit/compare/refs/tags/v0.3.0...v0.3.1

Changelog: https://github.com/chemelli74/aiocomelit/blob/0.3.1/CHANGELOG.md

fabaff committed:
python311Packages.amqp: add changelog to meta

fabaff committed:
python311Packages.amqp: update disabled

fabaff committed:
python311Packages.blinkpy: 0.22.2 -> 0.22.3

Diff: https://github.com/fronzbot/blinkpy/compare/refs/tags/v0.22.2...v0.22.3

Changelog: https://github.com/fronzbot/blinkpy/blob/refs/tags/v0.22.3/CHANGES.rst

fabaff committed:
python311Packages.evohome-async: 0.4.3 -> 0.4.4

Diff: https://github.com/zxdavb/evohome-async/compare/refs/tags/0.4.3...0.4.4

fabaff committed:
python311Packages.meshtastic: 2.2.11 -> 2.2.12

Diff: https://github.com/meshtastic/Meshtastic-python/compare/refs/tags/2.2.11...2.2.12

Changelog: https://github.com/meshtastic/python/releases/tag/2.2.12

fabaff committed:
python311Packages.lsassy: 3.1.8 -> 3.1.9

Diff: https://github.com/Hackndo/lsassy/compare/refs/tags/v3.1.8...v3.1.9

Changelog: https://github.com/Hackndo/lsassy/releases/tag/v3.1.9

fabaff committed:
python311Packages.rns: 0.6.5 -> 0.6.6

Diff: https://github.com/markqvist/Reticulum/compare/refs/tags/0.6.5...0.6.6

Changelog: https://github.com/markqvist/Reticulum/releases/tag/0.6.6

fabaff committed:
python311Packages.publicsuffixlist: 0.10.0.20231104 -> 0.10.0.20231105

fabaff committed:
python311Packages.pycep-parser: 0.4.1 -> 0.4.2

Diff: https://github.com/gruebel/pycep/compare/refs/tags/0.4.1...0.4.2

Changelog: https://github.com/gruebel/pycep/blob/0.4.2/CHANGELOG.md

fabaff committed:
python311Packages.plugwise: 0.34.4 -> 0.34.5

Diff: https://github.com/plugwise/python-plugwise/compare/refs/tags/v0.34.4...v0.34.5

Changelog: https://github.com/plugwise/python-plugwise/releases/tag/v0.34.5

fabaff committed:
python311Packages.pyoverkiz: 1.12.1 -> 1.12.2

Diff: https://github.com/iMicknl/python-overkiz-api/compare/refs/tags/v1.12.1...v1.12.2

Changelog: https://github.com/iMicknl/python-overkiz-api/releases/tag/v1.12.2

fabaff committed:
python311Packages.pysml: 0.1.0 -> 0.1.1

Diff: https://github.com/mtdcr/pysml/compare/refs/tags/0.1.0...0.1.1

fabaff committed:
python311Packages.tplink-omada-client: 1.3.5 -> 1.3.6

Changelog: https://github.com/MarkGodwin/tplink-omada-api/releases/tag/release%2Fv1.3.6

fabaff committed:
python311Packages.tldextract: 5.0.1 -> 5.1.0

Diff: https://github.com/john-kurkowski/tldextract/compare/refs/tags/5.0.1...5.1.0

Changelog: https://github.com/john-kurkowski/tldextract/blob/5.1.0/CHANGELOG.md

aaronjheng committed:
pb: 0.1.0 -> 0.2.0

thiagokokada committed:
Merge pull request #244298 from oxalica/remove-unused

{birdtray,isgx,mycrypto,osu-lazer,stretchly,tree-sitter}: remove oxalica as maintainer

r-ryantm committed:
python310Packages.gptcache: 0.1.41 -> 0.1.42

natsukium committed:
Merge pull request #265296 from r-ryantm/auto-update/kubectl-cnpg

kubectl-cnpg: 1.20.2 -> 1.21.0

pbsds committed:
Merge pull request #263015 from surfaceflinger/lunarclient-autoupdate

lunar-client: migrate to by-name and add updateScript

r-ryantm committed:
python310Packages.greeneye-monitor: 5.0 -> 5.0.1

secufoe committed:
rshim-user-space: make bfb-install optional

elohmeier committed:
mkpasswd: fix build with clang

fabaff committed:
Merge pull request #265792 from fabaff/boschshcpy-bump

python311Packages.boschshcpy: 0.2.73 -> 0.2.75

fabaff committed:
Merge pull request #265793 from fabaff/amqp-bump

python311Packages.amqp: 5.1.1 -> 5.2.0

fabaff committed:
Merge pull request #265794 from fabaff/aiocomelit-bump

python311Packages.aiocomelit: 0.3.0 -> 0.3.1

fabaff committed:
Merge pull request #265795 from fabaff/dvclive-bump

python311Packages.dvclive: 3.1.0 -> 3.2.0

teto committed:
Merge pull request #263320 from jfvillablanca/llm-ls

llm-ls: init at 0.4.0

r-ryantm committed:
python310Packages.grpcio-channelz: 1.59.0 -> 1.59.2

rnhmjoj committed:
Merge pull request #263138 from tomfitzhenry/hostapd-optional-managementframeprotection

nixos/hostapd: remove managementFrameProtection in favour of clearer default

r-ryantm committed:
python310Packages.grpcio-reflection: 1.59.0 -> 1.59.2

Artturin committed:
Merge pull request #254264 from amesgen/bazecor

bazecor: init at 1.3.6

amjoseph-nixpkgs commented on pkgs/development/libraries/ncurses/default.nix:
```diff
@@ -48,20 +48,18 @@ stdenv.mkDerivation (finalAttrs: {
         "/run/current-system/sw/share/terminfo" # NixOS
       ]}"
-  ];
+  ] ++ lib.optional (stdenv.buildPlatform != stdenv.hostPlatform) "--with-build-cc=${stdenv.cc.targetPrefix}cc";
```
I tried building `pkgsCross.aarch64-multiplatform.ncurses` with this PR and it didn't work.

But with the following change it did:

```suggestion
  ] ++ lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [
    "--with-build-cc=${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}cc"
  ];
```

Note that it's okay to use `buildPackages` here because this is isn't `depsFooBar`, `buildInputs`, or `nativeBuildInputs`.

klchen0112 committed:
librime: set darwin as support platforms

Kranzes committed:
Merge pull request #264658 from adamcstephens/kanidm/1.1.0-rc.15

kanidm: 1.1.0-beta.13 -> 1.1.0-rc.15

GaetanLepage committed:
python311Packages.jaxopt: 0.8.1 -> 0.8.2

Changelog: https://github.com/google/jaxopt/releases/tag/jaxopt-v0.8.2

wegank committed:
inkscape: fix runtime error on darwin

r-ryantm committed:
python310Packages.httpx-socks: 0.7.8 -> 0.8.0

natsukium committed:
Merge pull request #264463 from GaetanLepage/objax

python311Packages.objax: disable tests to fix build

wegank committed:
Merge pull request #245431 from leiserfg/staging

inkscape: 1.2.2 -> 1.3, lib2geom: 1.2.2 -> 1.3

NickCao committed:
Merge pull request #265842 from r-ryantm/auto-update/python310Packages.httpx-socks

python310Packages.httpx-socks: 0.7.8 -> 0.8.0

NickCao committed:
Merge pull request #265823 from r-ryantm/auto-update/python310Packages.greeneye-monitor

python310Packages.greeneye-monitor: 5.0 -> 5.0.1

NickCao committed:
Merge pull request #265813 from aaronjheng/pb

pb: 0.1.0 -> 0.2.0

NickCao committed:
Merge pull request #265811 from fabaff/pysml-bump

python311Packages.pysml: 0.1.0 -> 0.1.1

NickCao committed:
Merge pull request #265810 from fabaff/tplink-omada-client-bump

python311Packages.tplink-omada-client: 1.3.5 -> 1.3.6

mweinelt committed:
Merge pull request #265773 from r-ryantm/auto-update/python310Packages.faster-whisper

python310Packages.faster-whisper: 0.8.0 -> 0.9.0

NickCao committed:
Merge pull request #265808 from fabaff/pycep-parser-bump

python311Packages.pycep-parser: 0.4.1 -> 0.4.2

NickCao committed:
Merge pull request #265807 from fabaff/pyoverkiz-bump

python311Packages.pyoverkiz: 1.12.1 -> 1.12.2

NickCao committed:
Merge pull request #265806 from fabaff/rns-bump

python311Packages.rns: 0.6.5 -> 0.6.6

NickCao committed:
Merge pull request #265805 from fabaff/plugwise-bump

python311Packages.plugwise: 0.34.4 -> 0.34.5

NickCao committed:
Merge pull request #265804 from fabaff/meshtastic-bump

python311Packages.meshtastic: 2.2.11 -> 2.2.12

NickCao committed:
Merge pull request #265802 from fabaff/publicsuffixlist-bump

python311Packages.publicsuffixlist: 0.10.0.20231104 -> 0.10.0.20231105

r-ryantm committed:
lsp-plugins: 1.2.12 -> 1.2.13

NickCao committed:
Merge pull request #265799 from fabaff/evohome-async-bump

python311Packages.evohome-async: 0.4.3 -> 0.4.4

NickCao committed:
Merge pull request #265798 from fabaff/blinkpy-bump

python311Packages.blinkpy: 0.22.2 -> 0.22.3

NickCao committed:
Merge pull request #265809 from fabaff/tldextract-bump

python311Packages.tldextract: 5.0.1 -> 5.1.0

NickCao committed:
Merge pull request #265788 from r-ryantm/auto-update/python310Packages.gocardless-pro

python310Packages.gocardless-pro: 1.47.0 -> 1.48.0

NickCao committed:
Merge pull request #265791 from fabaff/alexapy-bump

python311Packages.alexapy: 1.27.7 -> 1.27.8

NickCao committed:
Merge pull request #265819 from r-ryantm/auto-update/python310Packages.gptcache

python310Packages.gptcache: 0.1.41 -> 0.1.42

NickCao committed:
Merge pull request #265772 from r-ryantm/auto-update/python310Packages.ezyrb

python310Packages.ezyrb: 1.3.0.post2309 -> 1.3.0.post2311

NickCao committed:
Merge pull request #265774 from r-ryantm/auto-update/python310Packages.fasttext-predict

python310Packages.fasttext-predict: 0.9.2.1 -> 0.9.2.2

NickCao committed:
Merge pull request #265775 from r-ryantm/auto-update/python310Packages.finvizfinance

python310Packages.finvizfinance: 0.14.6 -> 0.14.7

NickCao committed:
Merge pull request #265778 from r-ryantm/auto-update/python310Packages.flask-paginate

python310Packages.flask-paginate: 2023.10.8 -> 2023.10.24

NickCao committed:
Merge pull request #265784 from r-ryantm/auto-update/python310Packages.glcontext

python310Packages.glcontext: 2.4.0 -> 2.5.0

NickCao committed:
Merge pull request #265786 from honnip/update-pretendard

pretendard: 1.3.8 -> 1.3.9

NickCao committed:
Merge pull request #265803 from fabaff/lsassy-bump

python311Packages.lsassy: 3.1.8 -> 3.1.9

NickCao committed:
Merge pull request #265763 from r-ryantm/auto-update/python310Packages.dvclive

python310Packages.dvclive: 3.1.0 -> 3.2.0

NickCao committed:
Merge pull request #265766 from r-ryantm/auto-update/python310Packages.easydict

python310Packages.easydict: 1.10 -> 1.11

wegank committed:
Merge pull request #265824 from elohmeier/mkpasswd-clang

mkpasswd: fix build with clang

NickCao committed:
Merge pull request #265767 from kashw2/terragrunt

terragrunt: 0.53.0 -> 0.53.2

wegank committed:
Merge pull request #265831 from klchen0112/librime

librime: set darwin as support platforms

wegank committed:
Merge pull request #264619 from mdarocha/nuget-arm

nuget: allow aarch64-linux and darwin builds

natsukium committed:
Merge pull request #265835 from GaetanLepage/jaxopt

python311Packages.jaxopt: 0.8.1 -> 0.8.2

nikstur committed:
Merge pull request #261376 from secufoe/rshim-user-space-bfb-install

rshim-user-space: add bfb-install

pbsds committed:
Merge pull request #251858 from maxbrunet/feat/devpod

devpod: init at 0.4.1

Djabx committed:
google-cloud-sdk: 446.0.1 -> 452.0.1 (#264589)

kjeremy commented:
oops
